jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/JettyServerFactory.java

```diff
@@ -105,44 +108,50 @@ public static Server createServer(String name, Dictionary<String, Object> props,
                 }
             }
 
-            try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
+            if (jettyConfigurations != null)
             {
-                //create the server via config files
-                for (URL jettyConfiguration : jettyConfigurations)
+                try (ResourceFactory.Closeable resourceFactory = ResourceFactory.closeable())
                 {
-                    try
+                    //create the server via config files
+                    for (URL jettyConfiguration : jettyConfigurations)
                     {
-                        // Execute a Jetty configuration file
-                        XmlConfiguration config = new XmlConfiguration(resourceFactory.newResource(jettyConfiguration));
+                        try
+                        {
+                            Resource xmlresource = resourceFactory.newResource(jettyConfiguration);
+                            if (!Resources.isReadableFile(xmlresource))
+                                throw new FileNotFoundException("Unable to read: " + jettyConfiguration);
+                            // Execute a Jetty configuration file
+                            XmlConfiguration config = new XmlConfiguration(xmlresource);
 
-                        config.getIdMap().putAll(idMap);
-                        config.getProperties().putAll(properties);
+                            config.getIdMap().putAll(idMap);
+                            config.getProperties().putAll(properties);
 
-                        // #334062 compute the URL of the folder that contains the
-                        // conf file and set it as a property so we can compute relative paths
-                        // from it.
-                        String urlPath = jettyConfiguration.toString();
-                        int lastSlash = urlPath.lastIndexOf('/');
-                        if (lastSlash > 4)
+                            // #334062 compute the URL of the folder that contains the
+                            // conf file and set it as a property so we can compute relative paths
+                            // from it.
+                            String urlPath = jettyConfiguration.toString();
+                            int lastSlash = urlPath.lastIndexOf('/');
+                            if (lastSlash > 4)
+                            {
+                                urlPath = urlPath.substring(0, lastSlash);
+                                config.getProperties().put(PROPERTY_THIS_JETTY_XML_FOLDER_URL, urlPath);
+                            }
+
+                            Object o = config.configure();
+                            if (o instanceof Server configuredServer)
```

Comment:
So the code should probably be:
```suggestion
                            if (server == null && o instanceof Server configuredServer)
```

then at line 152:

```java
if (server == null)
{
   LOG.warn("No server configured by XML files");
}
```

Reply:
I don't think the server can be non-null at 141 as it is not assigned before then

Reply:
Done